### PR TITLE
Hierarchies-react: fix keyboard navigation issues

### PR DESCRIPTION
**Pull request by jasdom**
Reverted changes made with https://github.com/iTwin/presentation/pull/615/files as passing `onKeyDown` to `TreeNode` does not work as expected and causes issues with keyboard navigation as well as visual bugs.

### CODE REVIEW

File: packages/components/src/test/properties/inputs/UniquePropertyValuesSelector.test.tsx [hunk 1]
```diff
@@ -933,13 +933,16 @@ describe("UniquePropertyValuesSelector", () => {
...
-      expect(getDistinctValuesIteratorStub).to.be.calledThrice;
+      await waitFor(
+        () => {
+          expect(getDistinctValuesIteratorStub).to.be.calledThrice;
+          expect(queryByText("SearchedValue1")).to.not.be.null;
+          expect(queryByText("SearchedValue2")).to.not.be.null;
+          expect(queryByText("SearchedValue3")).to.not.be.null;
+          expect(queryAllByText(/SkippedValue/)).to.be.empty;
+        },
+        { timeout: 2000 },
```
Comment (saskliutas) [inline packages/components/src/test/properties/inputs/UniquePropertyValuesSelector.test.tsx:944]: Why is this needed?

Comment (jasdom) [reply packages/components/src/test/properties/inputs/UniquePropertyValuesSelector.test.tsx:944]: This test was occasionally failing. Did some debugging and it seems like the pages are requested correctly but sometimes it timeouts before the last page loaded.

**Review by saskliutas — APPROVED** [review]
